MvvmCross/Core/Core/Navigation/MvxNavigationService.cs

```diff
@@ -119,21 +120,23 @@ private static bool TryGetRoute(string url, out KeyValuePair<Regex, Type> entry)
             OnAfterNavigate(this, args);
         }
 
-        private async Task NavigateRoute(string path)
+        private async Task<IMvxViewModel> NavigateRoute(string path)
         {
             KeyValuePair<Regex, Type> entry;
```

Comment:
Dear MarcBruins,

Because TaskCompletionSource is not serializable, so I think { Task<TResult> Navigate<TViewModel,  TParameter, TResult>(TParameter param)  } will not work on Android, 
You can test it: open Settings ->Limit Background processes->select 'No background processes' -> reopen the app.
Return Task<Result> from NavigationService is great, but I think it is hard to implement it on Android.